api/tests/functional-tests/backend/core/test_label.py

```diff
@@ -584,68 +558,17 @@ def test_label_functions(
                 model_names=[model_name],
                 dataset_names=[dataset_name],
                 require_polygon=True,
-                task_types=[enums.TaskType.SEMANTIC_SEGMENTATION],
-            ),
-            ignore_groundtruths=True,
-        )
-        == set()
-    )
-
-    assert (
-        get_label_keys(
-            db,
-            schemas.Filter(
-                dataset_names=[dataset_name],
-                task_types=[enums.TaskType.CLASSIFICATION],
-            ),
-            ignore_predictions=True,
-        )
-        == set()
-    )
-    assert (
-        get_labels(
-            db,
-            schemas.Filter(
-                model_names=[model_name],
-                dataset_names=[dataset_name],
-                task_types=[enums.TaskType.CLASSIFICATION],
             ),
             ignore_groundtruths=True,
         )
         == set()
     )
 
-    assert get_label_keys(
-        db,
-        schemas.Filter(
-            dataset_names=[dataset_name],
-            task_types=[enums.TaskType.OBJECT_DETECTION],
-        ),
-        ignore_predictions=True,
-    ) == {"inssegk1", "inssegk2", "inssegk3"}
-
```

Comment:
Im not super happy losing the ability to query by task type. 

With the upcoming filters pr we could implement `Annotation.task_type == TaskType.OBJECT_DETECTION` as...

```
Annotation.is_instance == True & (
    Annotation.bounding_box.isnot(None)
    | Annotation.polygon.isnot(None)
    | Annotation.raster.isnot(None)
)
```

Reply:
Maybe it would be a good idea to leave task type in the database. We could rename it to datatype to support the text annotations.

This would significantly reduce the size of the generated sql statements.

Reply:
what's a practical example where we'd need to filter by task type? why can't we create a new filter (in a future PR) to check for `is_instance_segmentation=True`?

leaving `task_type` attached to `Annotations` via filters is confusing IMO

Reply:
From the standpoint of any of the getter function (`get_labels`, `get_datasets`, `get_models`, ...) this is problematic b/c you can no longer ask get me all the objects that have data for semantic segmenation or object detection.

Reply:
for discussion tomorrow: if I add `implied_task_types` to psql, can you add this filter back at your convenience?